framework/src/parser/CommandLine.C

```diff
@@ -248,3 +235,17 @@ CommandLine::setArgument<std::string>(std::stringstream & stream, std::string &
 {
   argument = stream.str();
 }
+
+HitCmpResult hitArgCmp(std::string arg, std::string app_name, int app_num)
+{
+  int arg_num = -1;
```

Comment:
OK - So I won't erase my comment about unsigned versus signed above because it definitely applies in MOOSE. Here this is a valid use of in since we are using it for a convenient "out of valid range" check. When using unsigned you can use `std::numeric_limits<T>::max()` or one of the related functions to get a good out of range value to check against.